### DISCUSSION TITLE
`d.multicall2` and custom views

### DISCUSSION THREAD

**Discussion by Rouneq** [rakshasa/rtorrent#1277] [Q&A] [answered]
According to the [Wiki](https://github.com/rakshasa/rtorrent/wiki/Common-Tasks-in-rTorrent), I can create a custom view definition in `.rtorrent.rc`. For example:

`schedule2 = filter_active,30,30,"view.filter = active,d.down.rate="`

Can this custom view be used with `d.multicall2`? If so, how?

I've put it in this exact view, restarted rtorrent, and calls using this view (or what I think the view name is) result in `Could not find view`. What am I missing here?

**Comment by kannibalox** [comment, marked as answer]
Looks like that page is lacking, the `active` view comes built-in to rtorrent so just changing the schedule is enough, but for a completely new view you'll have to use `view.add` first:
```bash
# Create the view, it's unfiltered by default
view.add = test
# Update with the defined filter every 30s
schedule2 = filter_test,30,30,"view.filter = test,d.down.rate="
```

**Reply by Rouneq** [reply]
Thanks @kannibalox. Took a bit of fiddling, but I did come up with something that worked. That said, I'm not sure it's 100%.

1) My rtorrent doesn't like end-of-line comments. I can't recall the exact error, but it was something like "incorrect config format"
2) The view config lines were positioned near the top. After directory/session path config, but before API config. Once rtorrent was restarted, `xmlrpc` only responded with fault codes.
3) Moved view config lines to the bottom of the file and restarted rtorrent. `xmlrpc` responds as expected.

Unfortunately, I'm not at a point where I can add more custom views to test if they will also work. Nor am I clear on why the API interface stopped working with the config lines located amongst the other lines.

**Reply by kannibalox** [reply]
Yep, the rtorrent.rc doesn't allow end-of-line comments, it's my fault for formatting it that way and I edited my reply.

I haven't seen the other behavior you're describing, those commands should either fail silently, or fail hard enough that rtorrent will throw an error on startup. I'd be interested to get a reproducible config if you have a chance, but if you got things working I'm not too worried about it.

**Reply by Rouneq** [reply]
I'd send you this directly, but it seems there isn't an easy way to do so. So I'll clutter up here. :)

```
# rTorrent configuration file

directory.default.set = /media/sdu/<username>/private/rtorrent/data
session.path.set = /media/sdu/<username>/private/rtorrent/work
directory.watch.added = /media/sdu/<username>/private/rtorrent/watch, load.start

# Downloading view
view.add = downloading
schedule2 = filter_downloading,1,1,"view.filter = downloading,d.down.rate="

# API
network.scgi.open_local = /media/sdu/<username>/private/rtorrent/.socket
execute2 = {sh,-c,/usr/bin/php /media/sdu/<username>/www/<username>.<server>.feralhosting.com/public_html/rutorrent/php/initplugins.php <username> &}

pieces.memory.max.set = 4G
pieces.preload.type.set = 0
pieces.hash.on_completion.set = no

# DHT
dht.mode.set = auto
schedule2 = dht_node, 3, 0, "dht.add_node=dht.libtorrent.org:25401"
schedule2 = dht_node, 3, 0, "dht.add_node=dht.transmissionbt.com"
schedule2 = dht_node, 3, 0, "dht.add_node=router.utorrent.com"
protocol.pex.set = yes

network.http.max_open.set = 1000
network.max_open_files.set = 1000
network.max_open_sockets.set = 1000

network.port_range.set = 32768-50000
network.port_random.set = no
protocol.encryption.set = allow_incoming, try_outgoing, enable_retry

encoding.add = utf8
system.file.allocate.set = yes
system.cwd.set = (directory.default)

throttle.max_peers.normal.set = 100
throttle.max_downloads.set = 100
throttle.max_uploads.set = 100

# Increasing these values does not normally increase speed but can negatively impact the server -- please be careful
throttle.max_downloads.global.set = 20
throttle.max_uploads.global.set = 40
```
